src/Common/MemoryTracker.cpp

```diff
@@ -165,6 +169,14 @@ void MemoryTracker::alloc(Int64 size)
         }
     }
 
+#ifdef MEMORY_TRACKER_DEBUG_CHECKS
+    if (unlikely(_memory_tracker_always_throw_logical_error_on_allocation))
```

Comment:
@tavplubix Maybe this check should be moved into `CurrentMemoryTracker::allocImpl`, so that it catches more issues, and not only when the allocation is huge enough (more then `max_untracked_memory`) ?

Reply:
>Seems like it worked for small allocations only because current_thread was nullptr in ThreadPool.

Indeed, since you are doing this from ThreadPool, which does not have memory tracker yet.

>However, can MemoryTracker be called directly without CurrentMemoryTracker?

Yes (i.e. see code with `getMemoryTracker`), but I don't see any code that call `free`/`alloc` w/o `CurrentMemoryTracker`, since these is done from the `Allocator.h`